### ISSUE TITLE
Render barriers bicycle access

### ISSUE THREAD

**Issue by Florimondable** [cyclosm/cyclosm-cartocss-style#219]
How to know if a barrier let bicycle pass ?
With `bicycle=yes`, so we could render them in blue.
And in red/orange when bicycle=no ?
I think the feature is realy usefull.

Examples
yes : https://www.openstreetmap.org/node/6864894893
no : https://www.openstreetmap.org/node/721443597

**Comment by Phyks** [comment]
Indeed. I would be more in favor of having grey by default and blue if bicycle can pass.

**Comment by Florimondable** [comment]
There are three states :
unknown, bicycle=yes, bicycle=no
so grey, blue, red/orange/yellow

**Comment by Phyks** [comment]
I would be in favor of considering unknown is the same as `bicycle=no` (as no precision likely means that bicycle are not allowed). Therefore, we could only consider:

* `bicycle=yes` in blue
* `bicycle=no` or unknown in grey

I'm not so much for having red/orange/yellow, this attracts too much the eye for something which is blocking.

**Comment by quasart** [comment]
`[barrier=gate][bicycle=yes]` is pretty useless to notice for a cyclist,
In my experience, it's the same effect than `[barrier=cycle_barrier]`, Maybe it is simpler to render them the same way.
https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dcycle_barrier

However, I would not make these barrier blue as blue means **good** for bike, but generally it is not that good : you need to slow down or to put your feet down.  In best cases, it's simply something you want to ignore.

**Comment by Phyks** [comment]
What about adopting the same strategy as for amenities. An amenity is rendered with full opacity but an `access=private|no` amenity is rendered with reduced opacity.

We could render barriers with full opacity (hard barriers) when bicycle access is forbidden. We could add them dimmed a bit (reduced opacity / grey instead of black) if they are `bicycle=yes` or bike-passable (e.g. cycle barrier).

What do you think?

**Comment by patman37** [comment]
We really must distinguish hard gates closed for bicycles which have a tag `access|bicycle=no|private`
I would colour them for example orange to make them stand out and also visible on a lower zoom level. On a long-distance cycle journey a single closed barrier can mean a long detour.

**Comment by mjray** [comment]
#549 was closed as a duplicate but suggested rendering slaloms and the much more obstructive A-barriers differently, and drawing the gap width above the barrier symbol.

At the moment, cyclosm renders A-barriers with a slalom symbol, which is misleading (and OSM suffers the same bug). A-barriers do not have a width:separation tag, but I am not sure how else they are distinguished.
